test/handshake_helper.c

```diff
@@ -1285,7 +1285,7 @@ static int pkey_type(EVP_PKEY *pkey)
 
 static int peer_pkey_type(SSL *s)
 {
-    X509 *x = SSL_get_peer_certificate(s);
+    X509 *x = SSL_get1_peer_certificate(s);
```

Comment:
```suggestion
    X509 *x = SSL_get0_peer_certificate(s);
```
and then remove the `X509_free(x)` below.